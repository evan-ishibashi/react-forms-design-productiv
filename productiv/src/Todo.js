import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({ id, title, description, priority, complete, update }) {

  function handleClick(){
    const newTodo = {
      id,
      title,
      description,
      priority,
      complete : !complete
    }
    update(newTodo);
  }

  return (
    <div className="Todo">
      <div>
        <div onClick={handleClick} >
          {complete ?
            <s><b>{title}</b></s> :
            <b>{title}</b>
          }
        </div>

        <small>(priority: {priority})</small>
      </div>
      <div><small>{description}</small></div>
    </div>
  );
}

export default Todo;
