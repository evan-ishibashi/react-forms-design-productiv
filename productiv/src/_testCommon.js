const TEST_TODOS = [
  {
    id: 1,
    title: "Test1",
    description: "Test1 - priority2",
    priority: 2,
  },
  {
    id: 2,
    title: "Test2",
    description: "Test2 -priority1",
    priority: 1,
  },
  {
    id: 3,
    title: "Test3",
    description: "Test3 - priority3",
    priority: 3,
  },
];


