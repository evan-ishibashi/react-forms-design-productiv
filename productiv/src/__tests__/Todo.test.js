import { fireEvent, render } from "@testing-library/react";
import Todo from "../Todo";
import {TEST_TODOS} from "../_testCommon";

const todo = TEST_TODOS[0];

//smoke test
test("Does it crash?", function (){
  render(<Todo {...todo}/>)
})

//snapshot test
test("Making a snapshot", function (){
  const { container } = render(<Todo {...todo} />);
  expect(container).toMatchSnapshot();
})


//actual tests
test("do we see a correct todo?", function (){
  const { queryByText } = render(<Todo {...todo} />);

  expect(queryByText("Test1 - priority2")).toContainHTML("<small>Test1 - priority2</small>");
  expect(queryByText("(priority: 2)")).toContainHTML("<small>(priority: 2)</small>");
});
