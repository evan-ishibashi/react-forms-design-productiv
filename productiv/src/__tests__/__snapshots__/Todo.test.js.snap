// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Making a snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Test1
      </b>
       
      <small>
        (priority: 
        2
        )
      </small>
    </div>
    <div>
      <small>
        Test1 - priority2
      </small>
    </div>
  </div>
</div>
`;
